### PR TITLE
DBus 1.12.8

### PR DESCRIPTION
**Pull request by Mno-hime**
Testing in MATE environment.

### CODE REVIEW

**Comment by pyhalov** [comment]
Have you rebuilt illumos-gate with new dbus?

**Comment by Mno-hime** [comment]
Yeah, it fails to build with:
```
/usr/include/dbus-1.0/dbus/dbus-address.h:74: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'void'
/usr/include/dbus-1.0/dbus/dbus-message.h:366: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'void'
/usr/include/dbus-1.0/dbus/dbus-connection.h:464: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'void'
/usr/include/dbus-1.0/dbus/dbus-pending-call.h:87: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'void'
/usr/include/dbus-1.0/dbus/dbus-server.h:115: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'void'
dmake: Warning: Command failed for target `pics/libhal.o'
```
`/usr/include/dbus-1.0/dbus/dbus-address.h`:
```
 74 static inline void
 75 dbus_clear_address_entries (DBusAddressEntry ***pointer_to_entries)
 76 {
 77   _dbus_clear_pointer_impl (DBusAddressEntry *, pointer_to_entries,
 78                             dbus_address_entries_free);
 79 }
```

https://cgit.freedesktop.org/dbus/dbus/tree/dbus/dbus-address.h?h=dbus-1.10
https://cgit.freedesktop.org/dbus/dbus/tree/dbus/dbus-address.h?h=dbus-1.12

Maybe our illumos-gcc doesn't like the  new stuff in DBus 1.12?

**Comment by pyhalov** [comment]
"static inline" in public header file is suspicious.

**Comment by Mno-hime** [comment]
Removed `inline` from those functions' definition and illumos-gate now builds, I am running it now. Test results are the same.

File: components/library/dbus/patches/10-remove-static-inline-from-headers.patch [hunk 1]
```diff
@@ -0,0 +1,55 @@
+--- dbus-1.12.8/dbus/dbus-address.h	2017-10-30 13:26:18.000000000 +0000
```
Comment (pyhalov) [inline components/library/dbus/patches/10-remove-static-inline-from-headers.patch]: Do I miss something? Have you added two patches to do the same?

Comment (Mno-hime) [reply components/library/dbus/patches/10-remove-static-inline-from-headers.patch]: Sorry about that, botched that patch. Should be fine now.

File: components/library/dbus/Makefile [hunk 2]
```diff
@@ -47,20 +49,28 @@ CONFIGURE_OPTIONS += --with-dbus-user=root
...
+test: $(TEST_32_and_64)
 
+REQUIRED_PACKAGES += SUNWcs
 REQUIRED_PACKAGES += developer/documentation-tool/doxygen
+REQUIRED_PACKAGES += developer/build/autoconf-archive
 REQUIRED_PACKAGES += library/expat
 REQUIRED_PACKAGES += library/gnome/yelp-tools
-REQUIRED_PACKAGES += SUNWcs
 REQUIRED_PACKAGES += system/library
+REQUIRED_PACKAGES += system/library/libdbus
```
Comment (pyhalov) [inline components/library/dbus/Makefile]: Please, remove it from REQUIRED_PACKAGES - component shouldn't depend on self

**Pull request merged**